File: node/register_domain.js
const { DNSimple, AuthenticationError } = require('dnsimple');

(async () => {
  const dnsimple = new DNSimple({
    baseUrl: 'https://api.sandbox.dnsimple.com',
    accessToken: process.env.TOKEN,
    userAgent: 'dnsimple-examples',
  });

  try {
    const identity = await dnsimple.identity.whoami();
    const accountId = identity.data.account.id;
    const domainName = process.env.DOMAIN;
    const contactId = process.env.CONTACT_ID;

    if (!domainName) {
      console.error('Please specify a domain name to register');
      return;
    }

    if (!contactId) {
      console.error('Please specify a contact ID to register the domain with');
      return;
    }

    const registrationAttributes = { registrant_id: contactId, auto_renew: true };
    const registration = await dnsimple.registrar.registerDomain(accountId, domainName, registrationAttributes);
    console.log(registration);
  } catch (err) {
    if (err instanceof AuthenticationError) {
      console.error('Authentication error. Check your token is correct for the sandbox environment.');
    } else {
      console.error(err);
    }
  }
})();

File: golang/cmd/register_domain/register_domain.go
// Usage: TOKEN=token go run cmd/register_domain/register_domain.go example.com
package main

import (
	"context"
	"fmt"
	"os"
	"strconv"

	"github.com/dnsimple/dnsimple-go/v6/dnsimple"
	"golang.org/x/oauth2"
)

func main() {
	oauthToken := os.Getenv("TOKEN")
	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: oauthToken})
	tc := oauth2.NewClient(context.Background(), ts)

	client := dnsimple.NewClient(tc)
	client.BaseURL = "https://api.sandbox.dnsimple.com"

	// get the current authenticated account (if you don't know who you are)
	whoamiResponse, err := client.Identity.Whoami(context.Background())
	if err != nil {
		fmt.Printf("Whoami() returned error: %v\n", err)
		os.Exit(1)
	}

	accountId := strconv.FormatInt(whoamiResponse.Data.Account.ID, 10)

	domainName := os.Args[1]
	fmt.Printf("Registering domain %v\n", domainName)
	contacts, err := client.Contacts.ListContacts(context.Background(), accountId, nil)

	if err != nil {
		fmt.Printf("ListContacts() returned error: %v\n", err)
		os.Exit(1)
	}

	if len(contacts.Data) == 0 {
		fmt.Printf("ListContacts() returned no contacts. Please create first a contact.")
		os.Exit(1)
	}

	contact := contacts.Data[0]

	registerDomainResponse, err := client.Registrar.RegisterDomain(context.Background(), accountId, domainName, &dnsimple.RegisterDomainInput{RegistrantID: int(contact.ID)})
	if err != nil {
		fmt.Printf("RegisterDomain() returned error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("%+v\n", registerDomainResponse.Data)
}
